### PR TITLE
Fix/publish only on pushes

### PR DESCRIPTION
**Pull request by Guillaumebeuzeboc**

### CODE REVIEW

File: .github/workflows/snap.yaml [hunk 1]
```diff
@@ -8,18 +8,23 @@ on:
...
 jobs:
   snap:
-    uses: ubuntu-robotics/snap_workflows/.github/workflows/snap.yaml@main
+    uses: ubuntu-robotics/snap_workflows/.github/workflows/snap.yaml@fix/default_branch_name
     secrets:
       store-login: ${{ secrets.STORE_LOGIN }}
     with:
       ubuntu-image: ubuntu-20.04
-      branch-name: noetic-devel
+      branch-name: ${{ inputs.branch-name }}
```
Comment (artivis) [inline .github/workflows/snap.yaml]: Isn't this input variable only available on `workflow_call`?

Comment (Guillaumebeuzeboc) [reply .github/workflows/snap.yaml]: Indeed, I will use an if condition instead.

Comment (Guillaumebeuzeboc) [reply .github/workflows/snap.yaml]: https://github.com/canonical/turtlebot3c-snap/actions/runs/8230696646/job/22504480753#step:1:43

Comment (Guillaumebeuzeboc) [reply .github/workflows/snap.yaml]: It worked with the if condition.

File: .github/workflows/snap.yaml [hunk 2]
```diff
@@ -8,18 +8,23 @@ on:
...
+   inputs:
+      branch-name:
+        required: false
+        type: string
+        default: ''
 
 jobs:
   snap:
-    uses: ubuntu-robotics/snap_workflows/.github/workflows/snap.yaml@main
+    uses: ubuntu-robotics/snap_workflows/.github/workflows/snap.yaml@fix/default_branch_name
```
Comment (artivis) [inline .github/workflows/snap.yaml]: Remember to change that back once the fix is merged.

File: .github/workflows/snap.yaml [hunk 3]
```diff
@@ -8,18 +8,23 @@ on:
...
     secrets:
       store-login: ${{ secrets.STORE_LOGIN }}
     with:
       ubuntu-image: ubuntu-20.04
-      branch-name: noetic-devel
+      branch-name: ${{ inputs.branch-name == '' && github.ref || inputs.branch-name }}
```
Comment (artivis) [inline .github/workflows/snap.yaml:24]: For future reference, this is a ternary operator-like hack. See e.g. the related [discussion](https://github.com/orgs/community/discussions/75928).

**Review by artivis — APPROVED** [review]

File: .github/workflows/snap.yaml [hunk 4]
```diff
@@ -8,18 +8,23 @@ on:
...
       snap-name: turtlebot3c
       snap-install-args: --dangerous
-      publish: true
+      publish: ${{ github.event_name == 'push' || github.event_name == 'workflow_call'}}
```
Comment (giusebar) [inline .github/workflows/snap.yaml:27]: why do we push also on the event that the workflow has been called, would that publish also for instance on a monthly check?

Comment (Guillaumebeuzeboc) [reply .github/workflows/snap.yaml:27]: Yes. And that is the idea. So the snaps available on the store are up to date.

**Comment by Guillaumebeuzeboc** [comment]
Back to main, ready for merge.

**Review by artivis — APPROVED** [review]

**Review by giusebar — APPROVED** [review]

**Pull request merged**